src/basic/unit-name.c

```diff
@@ -507,6 +514,68 @@ int unit_name_template(const char *f, char **ret) {
         return 0;
 }
 
+bool unit_name_is_hashed(const char *name) {
+        char *s;
+
+        if (!unit_name_is_valid(name, UNIT_NAME_PLAIN))
+                return false;
+
+        s = strrchr(name, '.');
+
+        if (s - name < UNIT_NAME_HASH_LENGTH_CHARS + 1)
+                return false;
+
+        s -= UNIT_NAME_HASH_LENGTH_CHARS + 1;
+        if (*s++ != '_')
+                return false;
+
+        for (size_t i = 0; i < UNIT_NAME_HASH_LENGTH_CHARS; i++)
+                if (unhexchar(s[i]) < 0)
+                        return false;
+
+        return true;
+}
+
+int unit_name_hash_long(const char *name, char **ret) {
+        _cleanup_free_ char *s = NULL, *hash = NULL;
+        char n[UNIT_NAME_MAX] = {}, *suffix;
+        le64_t h;
+        size_t len;
+
+        if (strlen(name) < UNIT_NAME_MAX)
+                return -EINVAL;
```

Comment:
maybe use a different error code here? just to signal the reader that this is a differnt kind of condition as the EINVAL cases below. The cases below after all are about truly invalid unit names. The one above just about overly long ones. maybe EMSGSIZE?